source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -592,8 +590,26 @@ bool JsonTranscoderFilter::maybeConvertGrpcStatus(Grpc::Status::GrpcStatus grpc_
     return false;
   }
 
+  response_headers_->Status()->value(Grpc::Utility::grpcToHttpStatus(grpc_status));
+
+  if (is_trailers_only_response) {
```

Comment:
not need to have this if.  non_trailer_only_response has been bailed out in line 560

Reply:
true, however I'm planning to remove that `if (!is_trailers_only_response) return` in next PR, which would fix the http connection manager so that it would allow to add data from the encodeTrailers handler.

Reply:
does that mean you're going to handle status to json conversion in non trailers_only_response case?

Reply:
Yes, I'll remove this in #8404 .

We can transcode status into json as long as upstream doesn't send its own reply body.
That includes a single headers frame with the end_stream flag (i.e. trailers-only), and two headers frames (headers+trailers).

Reply:
Removed it.